test/systemtests/kubeadm_test.go

```diff
@@ -572,14 +579,24 @@ func (k *kubePod) cleanupMaster() {
 }
 
 func getPodName(podRegex, nodeName string) (string, error) {
-	podNameCmd := `kubectl -n kube-system get pods -o wide | grep ` + podRegex + ` | grep ` + nodeName + ` | cut -d " " -f 1`
-	podName, err := k8sMaster.tbnode.RunCommandWithOutput(podNameCmd)
-	if err != nil {
-		logrus.Errorf("Couldn't fetch pod info on %s", nodeName)
-		return "", err
+	var err error
+	var podName string
+	for retry := 60; retry > 0; retry-- {
+		// only get running pods name
+		podNameCmd := `kubectl -n kube-system get pods -o wide | grep Running | grep ` + podRegex + ` | grep ` + nodeName + ` | cut -d " " -f 1`
```

Comment:
maybe there is a better kubectl query for this?  dunno if this helps, there is also all that json path stuff
```
kubectl get pods --field-selector=status.phase=Running
```

Reply:
yeah this is better, what i did this this line is just added `grep Running` piece ... I'd add it into the systest improvement PR

Reply:
I think this is not supported until 1.9
https://github.com/kubernetes/kubernetes/pull/50140/